### PR TITLE
Bump Dart SDK to 1.25.0-dev.9.0

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -40,7 +40,7 @@ class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox
   /// Proxy render boxes are rarely created directly because they simply proxy
   /// the render box protocol to [child]. Instead, consider using one of the
   /// subclasses.
-  RenderProxyBox([RenderBox child = null]) {
+  RenderProxyBox([RenderBox child = null]) { //ignore: avoid_init_to_null
```
Comment (a14n) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: If I remove `= null` I get the following issues from Analysis Server:

```
$ flutter analyze --flutter-repo --watch
  error • Invalid override. The type of 'RenderTransform.applyPaintTransform' ('(RenderBox, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(RenderObject, Matrix4) → void'). • packages/flutter/lib/src/rendering/proxy_box.dart:1665:3
  error • Invalid override. The type of 'RenderFittedBox.applyPaintTransform' ('(RenderBox, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(RenderObject, Matrix4) → void'). • packages/flutter/lib/src/rendering/proxy_box.dart:1799:3
  error • Invalid override. The type of 'RenderFractionalTranslation.applyPaintTransform' ('(RenderBox, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(RenderObject, Matrix4) → void'). • packages/flutter/lib/src/rendering/proxy_box.dart:1865:3
  error • Invalid override. The type of 'RenderFollowerLayer.applyPaintTransform' ('(RenderBox, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(RenderObject, Matrix4) → void'). • packages/flutter/lib/src/rendering/proxy_box.dart:3309:3
  error • Invalid override. The type of '_RenderSingleChildViewport.attach' ('(PipelineOwner) → void') isn't a subtype of 'AbstractNode.attach' ('(Object) → void'). • packages/flutter/lib/src/widgets/single_child_scroll_view.dart:231:3
  error • Invalid override. The type of '_RenderSliverPrototypeExtentList.attach' ('(PipelineOwner) → void') isn't a subtype of 'AbstractNode.attach' ('(Object) → void'). • packages/flutter/lib/src/widgets/sliver_prototype_extent_list.dart:151:3
6 issues found • analyzed 1145 files, 5.17 seconds
```

But directly with analyzer package:

```
$ flutter analyze --flutter-repo
Building flutter tool...
Analyzing 1144 files...
No analyzer warnings! (ran in 22.9s; 1 public member lacks documentation)
```

This looks like an issue in analysis server. /cc @bwilkerson @pq @danrubel

Comment (bwilkerson) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Is the version of analyzer being used by `flutter analyze` different than the version of server being used for `--watch`?

Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: I've definitely seen these kinds of errors before. Searching through the SDK bug list there's various bugs about "Invalid override", but I couldn't tell which were this specific issue.

Comment (pq) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: @bwilkerson : there shouldn't be analyzer skew since we're pulling in the analyzer from the SDK via a dependency_override:


```
dependency_overrides:
  analyzer:
    path: ../../bin/cache/dart-sdk/lib/analyzer
```

Comment (pq) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Having said that, I'm not sure that they're running with the same analysis _options_.  Maybe @danrubel can reassure us?

**Comment by pq** [comment]
Meta-comment `1.25.0-dev.9.0` is I think not slated to roll internally until Monday-ish.

File: packages/flutter/lib/src/rendering/proxy_box.dart (hunk 1 above)
Comment (a14n) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: @bwilkerson the difference between the results of `flutter analyze --flutter-repo --watch``and `flutter analyze --flutter-repo` is a secondary issue.

The main problem is that removing `= null` triggers 6 issues that look completely unrelated to the code changed. I tried to create a simpler reproduction without success.

Comment (bwilkerson) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: My first guess would be that this is related to strong mode and type inference, though I don't know why the presence or absence of a default value impacts it. Notice the difference in signatures:

'RenderTransform.applyPaintTransform' ('(**RenderBox**, Matrix4) → void') isn't a subtype of 'RenderProxyBoxMixin.applyPaintTransform' ('(**RenderObject**, Matrix4) → void')

@leafpetersen Any insights? Am I off-base with my guess?

Comment (leafpetersen) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: I don't think this is inference related.  I think it is related to strong mode override checking.  The fact that adding and removing `= null` changes the error looks like a bug to me.

Comment (leafpetersen) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: I'll try to extract out a repro.

Comment (leafpetersen) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Some notes:

I cannot reproduce this with the command line analyzer.  Running from bleeding edge, or from 1.25.0-dev.9.0, analyzing packages/flutter/lib/src/rendering/proxy_box.dart, I get no override errors with or without the `= null`.

Using Intellij, manually setting the dart sdk: 
  - If I point the dart sdk to 1.25-dev.9.0 I see the behavior described above.  I get a long list of overrider errors without `= null`, and some (but not all of them) go away if I add the `= null` back in.
 - If I point the dart sdk to bleeding edge, the variance caused by adding removing the unrelated code goes away, but I still see the errors.

I have extracted out the relevant class and override hierarchy into a single file example, but cannot reproduce with either the command line analyzer nor with Intellij (see code below).

The errors that I am seeing are consistent with a missing `covariant` on the base method, but the base method is marked `covariant`. 

If I remove the `covariant` from base method for applyPaintTransform, then the command line analyzer gives the same errors.

@bwilkerson I suspect that this is a bug specific to analysis server, and that specifically it may somehow be losing track of the `covariant` property.  I don't know what the code paths for the server are these days - this used to be computed during strong mode override inference in the task model.

My attempt to extract out a repo produced this code (which doesn't produce the error):

```dart
class Matrix4 {}

abstract class RenderObject {
void applyPaintTransform(covariant RenderObject child, Matrix4 transform) {
  }
}

abstract class RenderBox extends RenderObject {
  void applyPaintTransform(RenderObject child, Matrix4 transform) {}
}

abstract class RenderObjectWithChildMixin<ChildType extends RenderObject> extends RenderObject {
  ChildType _child;
  ChildType get child => _child;
  set child(ChildType value) {}
}

abstract class RenderProxyBoxMixin extends RenderBox with RenderObjectWithChildMixin<RenderBox> {
  @override
  void applyPaintTransform(RenderObject child, Matrix4 transform) { }

}

class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox>, RenderProxyBoxMixin {
  /// Creates a proxy render box.
  ///
  /// Proxy render boxes are rarely created directly because they simply proxy
  /// the render box protocol to [child]. Instead, consider using one of the
  /// subclasses.
  RenderProxyBox([RenderBox child = null]) {
    this.child = child;
  }
}


class RenderTransform extends RenderProxyBox {
@override
  void applyPaintTransform(RenderBox child, Matrix4 transform) {
  }
}
```

leafp-macbookpro:flutter leafp$ ../../bin/flutter doctor
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel master)
    • Flutter at /Users/leafp/src/flutter
    • Framework revision 00e135d879 (68 minutes ago), 2017-08-04 12:01:35 -0700
    • Engine revision 1de56a33d2
    • Tools Dart version 1.25.0-dev.7.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[-] Android Studio
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Ultimate Edition (version 2017.1.3)
    • Flutter plugin version 16.0
    • Dart plugin version 171.4249.16

[✓] Connected devices
    • None

leafp-macbookpro:flutter leafp$

Comment (leafpetersen) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Ok, I have a repro.  Splitting my minimal example into separate files reflecting the file layout of the actual code breaks the analysis server.  Issue filed here:

https://github.com/dart-lang/sdk/issues/30328

**Comment by Hixie** [comment]
LGTM

I wish we had a way to automatically catch places we need to update once upstream bugs are fixed.
